### PR TITLE
[JN-825] cohort rule evaluation

### PR DESCRIPTION
**Pull request by devonbush**
#### DESCRIPTION

This adds a Cohort rule evaluator.  Still only supporting basic =, !=, && and ||, but this probably covers 90% of the rules people need.  gradle seems to be complaining about an ambiguous execution order, I'll try to fix that in follow-on PRs

#### TO TEST:  *(simple manual steps for confirming core behavior -- used for pre-release checks)*

1. This doesn't have any functional changes to the application.  So just run the unit tests

### CODE REVIEW

File: core/src/main/java/bio/terra/pearl/core/rule/CohortRuleEvaluator.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+    public Object evaluateTerm(CohortRuleParser.TermContext ctx) throws RuleEvaluationException, RuleParsingException {
+        if (ctx.BOOLEAN() != null) {
+            return Boolean.parseBoolean(ctx.BOOLEAN().getText());
+        } else if (ctx.STRING() != null) {
+            String rawString = ctx.STRING().getText();
+            // trim off brackets and any whitespace immediately inside the brackets
+            return rawString.substring(1, rawString.length() - 1).trim();
+        } else if (ctx.NUMBER() != null) {
+            return Double.parseDouble(ctx.NUMBER().getText());
```
Comment (connorlbark) [inline core/src/main/java/bio/terra/pearl/core/rule/CohortRuleEvaluator.java:68]: is there any possible issue with it parsing all numbers as doubles? i worry about integer comparison / edge cases, but maybe that's unfounded since the tests seem to work OK.

File: core/src/test/java/bio/terra/pearl/core/rule/CohortRuleEvaluatorTests.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+import org.junit.jupiter.api.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+
+public class CohortRuleEvaluatorTests {
```
Comment (connorlbark) [inline core/src/test/java/bio/terra/pearl/core/rule/CohortRuleEvaluatorTests.java:11]: i like these tests, very clean

**Review by connorlbark — APPROVED** [review]
really cool stuff! excited to see where this goes.

File: core/src/main/java/bio/terra/pearl/core/rule/CohortRuleEvaluator.java (hunk 1 above)
Comment (devonbush) [reply core/src/main/java/bio/terra/pearl/core/rule/CohortRuleEvaluator.java:68]: We'd definitely have to worry if we were a bank -- but since we're just dealing with small, low-precision numbers, we should be fine.  FWIW our numbers mostly come from the frontend where everything is a double in Javascript

**Review by MatthewBemis — APPROVED** [review]
